File: dataset/human/python/file_8923.py
void main(){
	string s = readln().chomp().dup().reverse();
	string[] chk = ["remaerd", "maerd", "resare", "esare"];
	bool is_continue = true;
	while(is_continue){
		foreach(i,elm; chk){
			if(s.indexOf(elm)==0){
				s = s[elm.length..$];
				break;
			}
			// 最後まで見つからなければ終わり
			else if(i==3)is_continue = false;
		}
	}
	writefln("%s", s.length==0? "YES":"NO");
}

import std.stdio, std.conv, std.algorithm, std.numeric, std.string, std.math;

// 1要素のみの入力
T _scan(T= int)(){
	return to!(T)( readln().chomp() );
}

// 1行に同一型の複数入力
T[] _scanln(T = int)(){
	T[] ln;
	foreach(string elm; readln().chomp().split()){
		ln ~= elm.to!T();
	}
	return ln;
}

File: dataset/human/cpp/file_8923.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <map>
#include <queue>

bool f(std::vector<std::string> v1, std::vector<std::string> v2, int a, int b) {
    for (int k = a; k < v2.size() + a; k++) {
        for (int l = b; l < v2.size() + b; l++) {
            if (v1[k][l] != v2[k - a][l - b]) return false;
        }
    }
    return true;
}

int main() {
    int n, m;

    std::cin >> n >> m;

    std::vector<std::string> v1(n), v2(m);

    for (int i = 0; i < n; i++) {
        std::cin >> v1[i];
    }

    for (int i = 0; i < m; i++) {
        std::cin >> v2[i];
    }

    for (int i = 0; i <= n - m; i++) {
        for (int j = 0; j <= n - m; j++) {
            if (f(v1, v2, i, j)) {
                std::cout << "Yes" << std::endl;
                return 0;
            }
        }
    }
    std::cout << "No" << std::endl;
    return 0;
}
